tests/e2e/replication_experimental/multitenancy.py

```diff
@@ -0,0 +1,1508 @@
+    execute_and_fetch_all(main_cursor, "CREATE (:Node{on:'B'});")
+    execute_and_fetch_all(main_cursor, "CREATE (:Node{on:'B'});")
+
+    # 3/
+    interactive_mg_runner.kill(MEMGRAPH_INSTANCES_DESCRIPTION, "replica_2")
+    execute_and_fetch_all(main_cursor, "USE DATABASE memgraph;")
+    execute_and_fetch_all(main_cursor, "DROP DATABASE B;")
+    interactive_mg_runner.start(MEMGRAPH_INSTANCES_DESCRIPTION, "replica_2")
+
+    # 4/
+    expected_data = execute_and_fetch_all(main_cursor, "SHOW DATABASES;")
+
+    def retrieve_data(cursor):
+        return execute_and_fetch_all(cursor, "SHOW DATABASES;")
+
+    actual_data = mg_sleep_and_assert(expected_data, partial(retrieve_data, connection(7688, "replica_1").cursor()))
+    assert actual_data == expected_data
+    actual_data = mg_sleep_and_assert(expected_data, partial(retrieve_data, connection(7689, "replica_2").cursor()))
+    assert actual_data == expected_data
+
+
+def test_create_multitenancy_replication_restart_replica(connection):
+    # Goal: to show that replication can be established against REPLICA where a new databases
+    # needs replicating over
+    # 0/ Setup replication
+    # 1/ MAIN CREATE DATABASE A
+    # 2/ Write to MAIN A
+    # 3/ Validate replication of changes to A have arrived at REPLICA
+    # 4/ Start A transaction on replica 1, Use A on replica2
+    # 5/ Check that the drop replicated
+    # 6/ Validate that the transaction is still active and working and that the replica2 is not pointing to anything
+
+    # 0/
+    interactive_mg_runner.start_all(MEMGRAPH_INSTANCES_DESCRIPTION)
+    main_cursor = connection(7687, "main").cursor()
+
+    # 1/
+    execute_and_fetch_all(main_cursor, "CREATE DATABASE A;")
+
+    # 2/
+    execute_and_fetch_all(main_cursor, "USE DATABASE A;")
+    execute_and_fetch_all(main_cursor, "CREATE (:Node{on:'A'});")
+
+    # 3/
+    def retrieve_data():
+        execute_and_fetch_all(main_cursor, "USE DATABASE A;")
+        return set(execute_and_fetch_all(main_cursor, "SHOW REPLICAS;"))
+
+    expected_data = {
+        ("replica_1", "127.0.0.1:10001", "sync", 1, 0, "ready"),
+        ("replica_2", "127.0.0.1:10002", "async", 1, 0, "ready"),
+    }
+    actual_data = mg_sleep_and_assert(expected_data, retrieve_data)
+    assert actual_data == expected_data
+
+    # 4/
+    replica1_cursor = connection(7688, "replica").cursor()
+    replica2_cursor = connection(7689, "replica").cursor()
+
+    execute_and_fetch_all(replica1_cursor, "USE DATABASE A;")
+    execute_and_fetch_all(replica1_cursor, "BEGIN")
+    execute_and_fetch_all(replica2_cursor, "USE DATABASE A;")
+
+    execute_and_fetch_all(main_cursor, "USE DATABASE memgraph;")
+    execute_and_fetch_all(main_cursor, "DROP DATABASE A;")
+
+    # 5/
+    # TODO Remove this once there is a replica state for the system
+    execute_and_fetch_all(main_cursor, "CREATE DATABASE B;")
+    execute_and_fetch_all(main_cursor, "USE DATABASE B;")
```

Comment:
Do you need create database B query here? It seems it is not used

Reply:
It's so we need to do `SHOW REPLICA` on a particular db